ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -187,18 +189,29 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
     public incrementStep(scroll: boolean = true): void {
         const nextIndex = this.nextAvailableSectionIndex();
         if (nextIndex !== -1) {
-            if (scroll) {
-                this.scrollToTop();
-            }
             // enable any validation errors to be visible
             this.validationRequested = true;
             this.sendSectionAnalytics();
             this.currentSection.validate();
+
+            if (scroll) {
+                this.scrollToTop();
+            }
+
             if (this.currentSection.valid) {
                 this.resetValidationState();
-                this.currentSectionIndex = nextIndex;
-                this.visitedSectionIndexes[nextIndex] = true;
-                this.saveLastVisitedSectionIndex(nextIndex);
+
+                this.isPageBusy.pipe(
+                    tap(_ => this.isLoaded$.next(false)),
+                    debounceTime(250),
+                    filter(busy => !busy),
+                    take(1)
```

Comment:
#optional I see that 3 operators debounceTime, filter and take(1) are repeated in this file 2 times and in baseActivity 1 time. What do you think about creating the custom rxjs operator and reuse it?

Reply:
Extracted  a const.